src/EnergyPlus/OutputProcessor.cc

```diff
@@ -6606,16 +6594,9 @@ void UpdateMeterReporting(EnergyPlusData &state)
                                                                  state.dataIPShortCut->cAlphaFieldNames,
                                                                  state.dataIPShortCut->cNumericFieldNames);
 
-        varnameLen = index(Alphas(1), '[');
-        if (varnameLen != std::string::npos) Alphas(1).erase(varnameLen);
-
-        ReportFreq = determineFrequency(state, Alphas(2));
-
-        int meterIndex = findMeterIndexFromMeterName(Alphas(1));
-        if (meterIndex > 0) {
-            // MeterFileOnlyIndicator is true, CumulativeIndicator is true
-            SetInitialMeterReportingAndOutputNames(state, meterIndex, true, ReportFreq, true);
-        } else {
+        bool meterFileOnlyIndicator = true;
+        bool cumulativeIndicator = true;
+        if (!setupMeterFromMeterName(Alphas(1), Alphas(2), meterFileOnlyIndicator, cumulativeIndicator)) {
```

Comment:
Yep, OK, so all these calling points are a lot cleaner and DRYer.  This is good stuff, and I don't see any issues.  You are simply moving the repeated code to inside the lambda.  :+1: